components/TextFields/src/ContainedInputView/private/MDCTextControlStylePathDrawingUtils.h

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+@interface MDCTextControlStylePathDrawingUtils : NSObject
+
++ (void)addTopRightCornerToPath:(UIBezierPath *)path
+                      fromPoint:(CGPoint)point1
+                        toPoint:(CGPoint)point2
+                     withRadius:(CGFloat)radius;
++ (void)addBottomRightCornerToPath:(UIBezierPath *)path
```

Comment:
how are we expected to use this? Are all 4 corners suppose to be used together?

would a native API call work instead?
https://developer.apple.com/documentation/uikit/uibezierpath/1624356-bezierpathwithroundedrect?language=objc

Reply:
`+bezierPathWithRoundedRect:cornerRadius:` doesn't really meet our needs here. It wouldn't meet our needs for outlined textfields because the path for an outline in an outlined textfield has a break in it where the label goes. This method doesn't help us achieve that. It doesn't work for filled textfields because not all the corner radii in a filled background path, for example, are equal. The bottom corners are right angles. Given all this, are you okay with not using this method?